### ISSUE TITLE
[Bug]: Need to have all Rails endpoints with /api prefix

### ISSUE THREAD

**Issue by gurkanoluc** [Shopify/cli#1450]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

App

### Expected behavior

I have created a new Rails app using `npm init @shopify/app@latest` command, following https://shopify.dev/docs/apps/getting-started/create tutorial.

In the generated app, the `ShopifyApp::Engine` engine is mounted at `/api` prefix. I run this application behind a prefidefined ngrok URL and start the app and ngrok like below

```
npm run dev -- --tunnel-url https://foo-app.foo-apps.dev:5001
```

```
ngrok http --region=eu --hostname=foo-app.foo-apps.dev 5001
```

When I add new routes to the `routes.rb` file of the Rails application without `/api` prefix, I expect to be able to access them directly with `foo-app.foo-apps.dev/new-route-path`, however when I try to access the URL stays the same as `foo-app.foo-apps.dev/new-route-path` but the content of the page is what I have below. 

<img width="734" alt="Screenshot 2023-02-25 at 21 45 27" src="https://user-images.githubusercontent.com/163884/221380973-fd7a37d6-7d74-496d-befd-4272754e2898.png">

This new endpoint does not extend the `AuthenticatedController` class which comes with the Rails app provided by Shopify, it's purely for my own usage and it should not require any shop authentication.

For now I can get everything working by putting them behind `/api` prefix but as you can imagine this doesn't make sense for quite a few endpoints I have as they are not providing any API to any other service.

### Actual behavior

When I have a route which is not starting with `/api` prefix, I should be able to access to it and the Rails app should handle the request.

### Reproduction steps

1. Create a new Rails app with npm init @shopify/app@latest
2. Go to `config/routes.rb` file in your Rails app and add a new route like `get '/sidekiq', to: 'sidekiq#index'`
3. Try to access the https://<ngrok-url>/sidekiq

### Operating System

MacOS Ventura

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.23.0

### Shell

iTerm2 + zsh

### Node version (run `node -v` if you're not sure)

v18.0.0

### What language and version are you using in your application?

ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-darwin22]

**Comment by alvaro-shopify** [comment]
Hi @gurkanoluc . Thank you for rising up this issue. The problem is that the development configuration is using the `vite` config to redirect the backend requests through the frontend.

This is done here https://github.com/Shopify/shopify-frontend-template-react/blob/main/vite.config.js#L58-L61

So in case you create new endpoints in the backend with a different prefix you should add that rule to the `vite` config file. Once you deploy the backend regardless the frontend, you should not suffer this problem.

We are going to add a comment in the `routes.rb` that indicates that in case you need to add a route outside `/api` you should add the needed configuration in `web/frontend/vite.config.js`

Please, let me know if this works for you

**Comment by gurkanoluc** [comment]
Hi @alvaro-shopify,

Thanks for the explanation and potential solution. I have added a prefix that's different than API and it just worked! Thanks!

As you said for other users maybe adding this as a note to the `route.rb` file can be helpful.

To help others who might have the same issue after the update specific part in `vite.config.js` looks like this

```
server: {
    host: "localhost",
    port: process.env.FRONTEND_PORT,
    hmr: hmrConfig,
    proxy: {
      "^/(\\?.*)?$": proxyOptions,
      "^/api(/|(\\?.*)?$)": proxyOptions,
      "^/sidekiq(/|(\\?.*)?$)": proxyOptions,
    },
  },
```

`/sidekiq` is a prefix I added.

Thanks!
